tests/phpunit/tests/l10n/determineLocale.php

```diff
@@ -0,0 +1,258 @@
+<?php
+
+/**
+ * @group l10n
+ * @group i18n
+ *
+ * @covers ::determine_locale
+ */
+class Tests_L10n_DetermineLocale extends WP_UnitTestCase {
+	protected $locale;
+	protected static $user_id;
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$user_id = $factory->user->create(
+			array(
+				'role'   => 'administrator',
+				'locale' => 'userLocale',
+			)
+		);
+	}
+
+	public function tear_down() {
+		unset( $_SERVER['CONTENT_TYPE'], $_GET['_locale'], $_COOKIE['wp_lang'], $GLOBALS['pagenow'] );
+
+		parent::tear_down();
+	}
+
+	public function test_short_circuit_empty() {
```

Comment:
Add ticket and covers comments and everywhere.

Reply:
AFAIK `@ticket` annotations are only used when a test was added that covers a specific bug, for which the test is useful. Here I simply add tests for an existing function without changing any behavior or fixing bugs. So a ticket annotation is a bit pointless IMHO. And the origin of the code can be viewed in svn/git blame.

As for `@covers`, that's already added on the class itself.

Reply:
> > AFAIK @ticket annotations
> 
> I don't think that is how they are currently used. It just denotes which ticket the unit test added them.

That's how I used them in the past & have been told to use them, also matches the handbook: https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#annotations

> > As for @Covers, that's already added on the class itself.
> 
> Works for me, but other tests have not done this.

I literally copied this from the `getLocale.php` tests

Reply:
https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#annotations

> @covers – Used to annotate which function, method or class is actually being tested by a particular test. It is strongly recommended to add a @covers tag for every test.

Reply:
And `@covers` is added to every test automatically by adding it to the class once, which is also [the recommended way by PHPUnit](https://docs.phpunit.de/en/10.0/annotations.html#covers). I don't think this should be needlessly repeated in this file. Especially since there is precedence.